src/overlays/actors/ovl_En_Ge1/z_en_ge1.c

```diff
@@ -26,95 +73,744 @@ const ActorInit En_Ge1_InitVars = {
+        Flags_SetSwitch(globalCtx, (this->actor.params >> 8) & 0x3F);
+        this->cutsceneTimer = 50;
+        func_80106CCC(globalCtx);
+    } else if ((this->skelAnime.curFrame == 15.0f) || (this->skelAnime.curFrame == 19.0f)) {
+        Audio_PlayActorSound2(&this->actor, NA_SE_IT_HAND_CLAP);
+    }
+}
+
+void EnGe1_SetupOpen_GTGGuard(EnGe1* this, GlobalContext* globalCtx) {
+    if ((func_8010BDBC(&globalCtx->msgCtx) == 5) && (func_80106BC8(globalCtx) != 0)) {
+        this->actionFunc = EnGe1_Open_GTGGuard;
+        Animation_Change(&this->skelAnime, &D_0600A048, 1.0f, 0.0f, Animation_GetLastFrame(&D_0600A048), ANIMMODE_ONCE,
+                         -3.0f);
+        this->animation = &D_0600A048;
+        this->animFunc = EnGe1_StopFidget;
+        this->stateFlags &= ~(GE1_STATE_IDLE_ANIM);
+    }
+}
+
+void EnGe1_RefuseEntryTooPoor_GTGGuard(EnGe1* this, GlobalContext* globalCtx) {
+    if (func_8010BDBC(&globalCtx->msgCtx) == 2) {
+        this->actionFunc = EnGe1_CheckForCard_GTGGuard;
+        EnGe1_SetAnimationIdle(this);
+    }
+}
+
+void EnGe1_OfferOpen_GTGGuard(EnGe1* this, GlobalContext* globalCtx) {
+    this->stateFlags |= GE1_STATE_TALKING;
+    if ((func_8010BDBC(&globalCtx->msgCtx) == 4) && (func_80106BC8(globalCtx) != 0)) {
+        func_80106CCC(globalCtx);
+
+        switch (globalCtx->msgCtx.choiceIndex) {
+            case 0:
+                if (gSaveContext.rupees < 10) {
+                    func_8010B720(globalCtx, 0x6016);
+                    this->actionFunc = EnGe1_RefuseEntryTooPoor_GTGGuard;
+                } else {
+                    Rupees_ChangeBy(-10);
+                    func_8010B720(globalCtx, 0x6015);
+                    this->actionFunc = EnGe1_SetupOpen_GTGGuard;
+                }
+                break;
+            case 1:
+                this->actionFunc = EnGe1_CheckForCard_GTGGuard;
+                EnGe1_SetAnimationIdle(this);
+                break;
+        }
+    }
+}
+
+void EnGe1_RefuseOpenNoCard_GTGGuard(EnGe1* this, GlobalContext* globalCtx) {
+    this->stateFlags |= GE1_STATE_TALKING;
+    if (func_8002F334(&this->actor, globalCtx)) {
+        this->actionFunc = EnGe1_CheckForCard_GTGGuard;
+        EnGe1_SetAnimationIdle(this);
+    }
+}
+
+void EnGe1_CheckForCard_GTGGuard(EnGe1* this, GlobalContext* globalCtx) {
+    if (CHECK_QUEST_ITEM(QUEST_GERUDO_CARD)) {
+        EnGe1_SetTalkAction(this, globalCtx, 0x6014, 100.0f, EnGe1_OfferOpen_GTGGuard);
+    } else { //! @bug This branch is inaccessible in normal gameplay since this function it is unreachable without
+             //! obtaining the card in the first place.
+        EnGe1_SetTalkAction(this, globalCtx, 0x6013, 100.0f, EnGe1_RefuseOpenNoCard_GTGGuard);
+    }
+}
+
+// Gate Operator functions
+
+void EnGe1_WaitGateOpen_GateOp(EnGe1* this, GlobalContext* globalCtx) {
+    this->stateFlags |= GE1_STATE_TALKING;
+
+    if ((func_8010BDBC(&globalCtx->msgCtx) == 5) && (func_80106BC8(globalCtx) != 0)) {
```

Comment:
```suggestion
    if ((func_8010BDBC(&globalCtx->msgCtx) == 5) && func_80106BC8(globalCtx)) {
```
Pretty sure this is boolean.